api/src/main/java/cz/fi/muni/pa165/dto/UserAuthenticateDTO.java

```diff
@@ -0,0 +1,32 @@
+package cz.fi.muni.pa165.dto;
+
+/**
+ * Created by Erik Macej 433744 , on 19.11.16.
+ *
+ * @author Erik Macej 433744
+ */
+public class UserAuthenticateDTO {
+
+    private String passwordHash;
```

Comment:
The client should not implement hashing. It is less secure and problematic for future refactoring.
It should be done on the server side; i.e. exchange just passwords instead.